save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt

```diff
@@ -0,0 +1,76 @@
+@file:Suppress("HEADER_MISSING_IN_NON_SINGLE_CLASS_FILE")
+
+package org.cqfn.save.frontend.components.basic
+
+import org.w3c.dom.events.Event
+import react.RBuilder
+import react.dom.*
+
+import kotlinx.html.js.onChangeFunction
+
+/**
+ * @param form
+ * @param validInput
+ * @param classes
+ * @param text
+ * @param onChangeFun
+ * @param elements
+ * @param initialValue
+ * @return div with an input form
+ */
+@Suppress(
+    "TOO_MANY_PARAMETERS",
+    "TOO_LONG_FUNCTION",
+    "LongParameterList"
+)
+internal fun RBuilder.selectFormRequired(
+    form: InputTypes,
+    validInput: Boolean,
+    classes: String,
+    text: String,
+    elements: List<String>,
+    initialValue: String?,
```

Comment:
Is it currently selected item? It should be in the state of this component, since you've already extracted it into a separate function. Otherwise it makes already large state of `ProjectView` even more complex :(

Reply:
Fixed